yt/frontends/swift/data_structures.py

```diff
@@ -0,0 +1,209 @@
+        """
+        units = self._get_info_attributes("Units")
+
+        if self.cosmological_simulation == 1:
+            only_on_root(
+                    mylog.info, "Assuming length units are in cm (comoving)")
+            self.length_unit = self.quan(
+                float(units["Unit length in cgs (U_L)"]), "cmcm")
+        else:
+            only_on_root(
+                    mylog.info, "Assuming length units are in cm (physical)")
+            self.length_unit = self.quan(
+                float(units["Unit length in cgs (U_L)"]), "cm")
+
+        self.mass_unit = self.quan(
+            float(units["Unit mass in cgs (U_M)"]), "g")
+        self.time_unit = self.quan(
+            float(units["Unit time in cgs (U_t)"]), "s")
+        self.temperature_unit = self.quan(
+            float(units["Unit temperature in cgs (U_T)"]), "K")
+
+        return
+
+    def _get_info_attributes(self, dataset):
+        """
+        Gets the information from a header-style dataset and returns it as a
+        python dictionary.
+
+        Example: self._get_info_attributes(header) returns a dictionary of all
+        of the information in the Header.attrs.
+        """
+
+        with h5py.File(self.filename, "r") as handle:
+            header = dict(handle[dataset].attrs)
+
+        return header
+
+    def _parse_parameter_file(self):
+        """
+        Parse the SWIFT "parameter file" -- really this actually reads info
+        from the main HDF5 file as everything is replicated there and usually
+        parameterfiles are not transported.
+
+        The header information from the HDF5 file is stored in an un-parsed
+        format in self.parameters should users wish to use it.
+        """
+
+        self.unique_identifier = uuid4()
+
+        # Read from the HDF5 file, this gives us all the info we need. The rest
+        # of this function is just parsing.
+        header = self._get_info_attributes("Header")
+        runtime_parameters = self._get_info_attributes("RuntimePars")
+
+        policy = self._get_info_attributes("Policy")
+        # These are the parameterfile parameters from *.yml at runtime
+        parameters = self._get_info_attributes("Parameters")
+
+        # Not used in this function, but passed to parameters
+        hydro = self._get_info_attributes("HydroScheme")
+        subgrid = self._get_info_attributes("SubgridScheme")
+
+        self.domain_right_edge = header["BoxSize"]
+        self.domain_left_edge = np.zeros_like(self.domain_right_edge)
+
+        self.dimensionality = int(header["Dimension"])
+
+        # SWIFT is either all periodic, or not periodic at all
+        periodic = int(runtime_parameters["PeriodicBoundariesOn"])
+
+        if periodic:
+            self.periodicity = [True] * self.dimensionality
+        else:
+            self.periodicity = [False] * self.dimensionality
+
+        self.current_time = float(header["Time"])
```

Comment:
does this end up with units attached after the dataset is fully loaded? if not it probably should have units explicitly attached here

Reply:
That's a good point - I should check that. I should add a test for time/velocity units

Reply:
Yes this time should be in the same units as the rest of the file (based on the I/O units).